src/start/syntax.md

```diff
@@ -1 +1,438 @@
+| `(fn [] => 0 | x::_ => x + 1)`                  | `int list`         | Yes               |
+| `(fn (0,b) => true andalso b)`                  | `int * bool`       | No                |
+| `(fn (0,b) => true andalso b | (n,_) => false)` | `int * bool`       | Yes               |
+
+Using a wildcard for the first clause's *entire* pattern produces an exhaustive match.
+
+#### Recursive Function Binding
+
+The `rec` reserved word enables a lambda expression to self-reference within its body.
+The `fun` reserved word allows self-reference by default.
+The clause patterns and expressions in `fun` clauses are separated by `=` instead of `=>`.
+```sml
+val rec length : int list -> int = fn [] => 0 | _::xs => 1 + length xs
+
+fun length ([]    : int list) : int = 0
+  | length (_::xs : int list) : int = 1 + length xs
+```
+
+As before, *both* `length` bindings have the same value. Don't forget about the lambda!
+
+|     Expression     |                 Value                 |       Type        |
+|:------------------:|:-------------------------------------:|:-----------------:|
+| `length`           | `fn [] => 0 | _::xs => 1 + length xs` | `int list -> int` |
+| `length []`        | `0`                                   | `int`             |
+| `length [1,2,3,4]` | `4`                                   | `int`             |
+
+#### Conditional Expressions
+
+Require a condition that evaluates to `true` or `false`, after the `if`.
+Two expressions of the same type — one for the `then`-branch, one for the `else`-branch.
+Note that `if`-`then`-`else` expressions only evaluate one of its two branches — the one it takes.
+
+|                              Expression                               | Evaluates To |
+|:---------------------------------------------------------------------:|:------------:|
+| `if 0 <> 1 then "foo" else "bar"`                                     | `"foo"`      |
+| `if 0 = 1 then (if true then 1 else 2) else (if false then 3 else 4)` | `4`          |
+| `if true then 1 else (1 div 0)`                                       | `1`          |
+| `if false then (1 div 0) else 0`                                      | `0`          |
+
+#### op
+
+The `op` keyword converts a binary infix operator to binary prefix operation. Priorities are kept the same as before.
+
+|       Expression       | Evaluates To |
+|:----------------------:|:------------:|
+| `(op *) (8,3)`         | `24`         |
+| `(op +) (3,4)`         | `7`          |
+| `(op ^) ("foo","bar")` | `"foobar"`   |
+| `(op ::) (1,[2,3,4])`  | `[1,2,3,4]`  |
+| `(op @) ([1,2],[3,4])` | `[1,2,3,4]`  |
+
+#### as
+
+If convenient, we can use the `as` keyword between a variable and a structured pattern to reference a structured value both as a whole and by its constituents. The pattern to the left of `as` must be a variable. It can be nested. It is always part of a pattern.
+```sml
+val tuple as (a,b) : int * int = (1,2)
+```
+
+| Expression | Evaluates To |
```

Comment:
```suggestion
| Variable Name | Bound To |
```
For the three tables in the "as" section, I think it would help to emphasize that `as` controls how pattern matching/binding works

Reply:
Thank you for suggesting that — done!